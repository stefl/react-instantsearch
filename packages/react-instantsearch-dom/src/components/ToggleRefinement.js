import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import createClassNames from '../core/createClassNames';

const cx = createClassNames('ToggleRefinement');

const ToggleRefinement = ({ currentRefinement, label, refine, className }) => (
  <div className={classNames(cx(''), className)}>
    <label className={cx('label')}>
      <input
        className={cx('checkbox')}
        type="checkbox"
        checked={currentRefinement}
        onChange={event => refine(event.target.checked)}
      />
      <span className={cx('labelText')}>{label}</span>
    </label>
  </div>
);

ToggleRefinement.propTypes = {
  currentRefinement: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
  refine: PropTypes.func.isRequired,
  className: PropTypes.string,
};

ToggleRefinement.defaultProps = {
  className: '',
};

export default ToggleRefinement;
