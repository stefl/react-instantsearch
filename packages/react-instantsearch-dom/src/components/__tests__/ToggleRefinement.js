import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ToggleRefinement from '../ToggleRefinement';

Enzyme.configure({ adapter: new Adapter() });

describe('ToggleRefinement', () => {
  const defaultProps = {
    currentRefinement: true,
    label: 'toggle the refinement',
    refine: () => {},
  };

  it('expect to render with a positive currentRefinement', () => {
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(<ToggleRefinement {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with a negative currentRefinement', () => {
    const props = {
      ...defaultProps,
      currentRefinement: false,
    };

    const wrapper = shallow(<ToggleRefinement {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with custom className', () => {
    const props = {
      ...defaultProps,
      className: 'MyCustomToggleRefinement',
    };

    const wrapper = shallow(<ToggleRefinement {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to call refine onChange', () => {
    const props = {
      ...defaultProps,
      refine: jest.fn(),
    };

    const wrapper = shallow(<ToggleRefinement {...props} />);

    expect(props.refine).not.toHaveBeenCalled();

    wrapper.find('input[type="checkbox"]').simulate('change', {
      target: {
        checked: false,
      },
    });

    expect(props.refine).toHaveBeenCalledWith(false);
  });
});
