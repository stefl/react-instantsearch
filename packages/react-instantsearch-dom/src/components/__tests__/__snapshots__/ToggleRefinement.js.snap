// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleRefinement expect to render with a negative currentRefinement 1`] = `
<div
  className="ais-ToggleRefinement"
>
  <label
    className="ais-ToggleRefinement-label"
  >
    <input
      checked={false}
      className="ais-ToggleRefinement-checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ais-ToggleRefinement-labelText"
    >
      toggle the refinement
    </span>
  </label>
</div>
`;

exports[`ToggleRefinement expect to render with a positive currentRefinement 1`] = `
<div
  className="ais-ToggleRefinement"
>
  <label
    className="ais-ToggleRefinement-label"
  >
    <input
      checked={true}
      className="ais-ToggleRefinement-checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ais-ToggleRefinement-labelText"
    >
      toggle the refinement
    </span>
  </label>
</div>
`;

exports[`ToggleRefinement expect to render with custom className 1`] = `
<div
  className="ais-ToggleRefinement MyCustomToggleRefinement"
>
  <label
    className="ais-ToggleRefinement-label"
  >
    <input
      checked={true}
      className="ais-ToggleRefinement-checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="ais-ToggleRefinement-labelText"
    >
      toggle the refinement
    </span>
  </label>
</div>
`;
